app/code/Magento/InventoryConfigurableProduct/Ui/DataProvider/Product/Form/Modifier/InventoryConfigurablePanel.php

```diff
@@ -52,10 +119,42 @@ private function getQuantityContainerConfig(): array
      */
     public function modifyMeta(array $meta)
     {
-        $matrix = $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children'][ConfigurablePanel::CONFIGURABLE_MATRIX];
-        $matrix['children']['record']['children']['quantity_container'] = $this->getQuantityContainerConfig();
-        $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children'][ConfigurablePanel::CONFIGURABLE_MATRIX] = $matrix;
+        if ($this->isSingleSourceMode->execute() === false) {
+            $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children']
+            [ConfigurablePanel::CONFIGURABLE_MATRIX]['children']
+            ['record']['children']['quantity_per_source_container'] = $this->getQuantityContainerConfig();
+
+            unset($meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children']
+                [ConfigurablePanel::CONFIGURABLE_MATRIX]['children']
+                ['record']['children']['quantity_container']);
+        }
 
         return $meta;
     }
+
+    /**
+     * @param string $sku
+     *
+     * @return array
+     */
+    private function getQuantityPerSource(string $sku): array
```

Comment:
Pls, move this in separate model (without interface)

Reply:
Done.